python_toolbox/get_colmap_SfM_logfile.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python
+#
+# ----------------------------------------------------------------------------
+# -                   TanksAndTemples Website Toolbox                        -
+# -                    http://www.tanksandtemples.org                        -
+# ----------------------------------------------------------------------------
+# The MIT License (MIT)
+#
+# Copyright (c) 2017
+# Arno Knapitsch <arno.knapitsch@gmail.com >
+# Jaesik Park <syncle@gmail.com>
+# Qian-Yi Zhou <Qianyi.Zhou@gmail.com>
+# Vladlen Koltun <vkoltun@gmail.com>
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+# ----------------------------------------------------------------------------
+#
+# Python script to convert COLMAP SfM data into the tanksandtempls log file format
+# Example: python get_colmap_SfM_logfile.py [COLMAP SfM file] [output log-filename] [folder of the input images] [image format]
+
+import sys
+import glob
+import numpy as np
+from numpy import matrix
+
+
+def write_trajectory(traj, filename):
+    with open(filename, 'w') as f:
+        for x in traj:
+            p = x.pose.tolist()
+            f.write(' '.join(map(str, x.metadata)) + '\n')
+            f.write('\n'.join(' '.join(map('{0:.12f}'.format, p[i])) for i in range(4)))
+            f.write('\n')
+            
+            
+def write_SfM_log(T, i_map, filename):
+    with open(filename, 'w') as f:
+        ii=0
+        for t in T:
+            p = t.tolist()
+            f.write(' '.join(map(str, i_map[ii])) + '\n')
+            f.write('\n'.join(' '.join(map('{0:.12f}'.format, p[i])) for i in range(4)))
+            f.write('\n')
+            ii = ii + 1
+                        
```

Comment:
These functions seems identical or similar to the functions used in get_VSFM_SfM_logfile.py. Can you make separate Python file for these functions and import them here?